[2024-12-23T08:43:03.619+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-12-23T08:43:03.658+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: streaming_pipeline.cdc_spark_streaming_to_datalake manual__2024-12-23T08:42:59.312112+00:00 [queued]>
[2024-12-23T08:43:03.671+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: streaming_pipeline.cdc_spark_streaming_to_datalake manual__2024-12-23T08:42:59.312112+00:00 [queued]>
[2024-12-23T08:43:03.672+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-12-23T08:43:03.691+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): cdc_spark_streaming_to_datalake> on 2024-12-23 08:42:59.312112+00:00
[2024-12-23T08:43:03.700+0000] {standard_task_runner.py:63} INFO - Started process 962 to run task
[2024-12-23T08:43:03.704+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'streaming_pipeline', 'cdc_spark_streaming_to_datalake', 'manual__2024-12-23T08:42:59.312112+00:00', '--job-id', '63', '--raw', '--subdir', 'DAGS_FOLDER/pipline_stream.py', '--cfg-path', '/tmp/tmpx3sdlydv']
[2024-12-23T08:43:03.707+0000] {standard_task_runner.py:91} INFO - Job 63: Subtask cdc_spark_streaming_to_datalake
[2024-12-23T08:43:03.770+0000] {task_command.py:426} INFO - Running <TaskInstance: streaming_pipeline.cdc_spark_streaming_to_datalake manual__2024-12-23T08:42:59.312112+00:00 [running]> on host 0b4a5bf62610
[2024-12-23T08:43:03.897+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='streaming_pipeline' AIRFLOW_CTX_TASK_ID='cdc_spark_streaming_to_datalake' AIRFLOW_CTX_EXECUTION_DATE='2024-12-23T08:42:59.312112+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-23T08:42:59.312112+00:00'
[2024-12-23T08:43:03.899+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-12-23T08:43:03.921+0000] {logging_mixin.py:188} INFO - Checking JAR files...:  True
[2024-12-23T08:43:22.284+0000] {spark_streaming_to_dl.py:123} INFO - Spark session successfully created!
[2024-12-23T08:43:22.297+0000] {spark_streaming_to_dl.py:157} INFO - MinIO configuration is created successfully
[2024-12-23T08:43:24.901+0000] {spark_streaming_to_dl.py:176} INFO - Initial dataframe created successfully!
[2024-12-23T08:43:26.129+0000] {spark_streaming_to_dl.py:234} INFO - Final dataframe created successfully!
[2024-12-23T08:43:26.130+0000] {spark_streaming_to_dl.py:242} INFO - Streaming is being started...
[2024-12-23T08:54:46.429+0000] {local_task_job_runner.py:310} WARNING - State of this instance has been externally set to None. Terminating instance.
[2024-12-23T08:54:46.432+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-12-23T08:54:46.437+0000] {process_utils.py:132} INFO - Sending 15 to group 962. PIDs of all processes in the group: [964, 962]
[2024-12-23T08:54:46.438+0000] {process_utils.py:87} INFO - Sending the signal 15 to group 962
[2024-12-23T08:54:46.439+0000] {taskinstance.py:2611} ERROR - Received SIGTERM. Terminating subprocesses.
[2024-12-23T08:54:46.441+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-12-23T08:54:46.461+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/stream_processing/spark_streaming_to_dl.py", line 269, in run_all
    start_steaming(df_final, args.bucket_name, args.year, args.month)
  File "/opt/airflow/dags/stream_processing/spark_streaming_to_dl.py", line 251, in start_steaming
    .start()
     ^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/streaming/readwriter.py", line 1527, in start
    return self._sq(self._jwrite.start())
                    ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/java_gateway.py", line 1321, in __call__
    answer = self.gateway_client.send_command(command)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/java_gateway.py", line 1038, in send_command
    response = connection.send_command(command)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/py4j/clientserver.py", line 511, in send_command
    answer = smart_decode(self.stream.readline()[:-1])
                          ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/socket.py", line 708, in readinto
    return self._sock.recv_into(b)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 2613, in signal_handler
    raise AirflowTaskTerminated("Task received SIGTERM signal")
airflow.exceptions.AirflowTaskTerminated: Task received SIGTERM signal
[2024-12-23T08:54:46.477+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=streaming_pipeline, task_id=cdc_spark_streaming_to_datalake, run_id=manual__2024-12-23T08:42:59.312112+00:00, execution_date=20241223T084259, start_date=20241223T084303, end_date=20241223T085446
[2024-12-23T08:54:46.524+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 63 for task cdc_spark_streaming_to_datalake ((psycopg2.errors.ForeignKeyViolation) insert or update on table "task_fail" violates foreign key constraint "task_fail_ti_fkey"
DETAIL:  Key (dag_id, task_id, run_id, map_index)=(streaming_pipeline, cdc_spark_streaming_to_datalake, manual__2024-12-23T08:42:59.312112+00:00, -1) is not present in table "task_instance".

[SQL: INSERT INTO task_fail (task_id, dag_id, run_id, map_index, start_date, end_date, duration) VALUES (%(task_id)s, %(dag_id)s, %(run_id)s, %(map_index)s, %(start_date)s, %(end_date)s, %(duration)s) RETURNING task_fail.id]
[parameters: {'task_id': 'cdc_spark_streaming_to_datalake', 'dag_id': 'streaming_pipeline', 'run_id': 'manual__2024-12-23T08:42:59.312112+00:00', 'map_index': -1, 'start_date': datetime.datetime(2024, 12, 23, 8, 43, 3, 658787, tzinfo=Timezone('UTC')), 'end_date': datetime.datetime(2024, 12, 23, 8, 54, 46, 474741, tzinfo=Timezone('UTC')), 'duration': 702}]
(Background on this error at: https://sqlalche.me/e/14/gkpj); 962)
[2024-12-23T08:54:46.975+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=962, status='terminated', exitcode=1, started='08:43:03') (962) terminated with exit code 1
[2024-12-23T08:54:47.230+0000] {process_utils.py:80} INFO - Process psutil.Process(pid=964, status='terminated', started='08:43:03') (964) terminated with exit code None
